pypeit/pypeit.py

```diff
@@ -417,102 +303,233 @@ def reduce_exposure(self, frame, bgframe=None, std_frame=None, reuse_masters=Fal
+        initial_sky = \
+            self.sciI.global_skysub(self.caliBrate.tilts_dict['tilts'], skymask=skymask_init,
+                                    std=self.std_redux, maskslits=self.maskslits, show=self.show)
+
+        if not self.std_redux:
+            # Object finding, second pass on frame *with* sky subtraction. Show here if requested
+            self.sobjs_obj, self.nobj, self.skymask = \
+                self.find_objects(sciimg - initial_sky, std=self.std_redux, ir_redux=self.ir_redux, std_trace=std_trace, snr_trim=True,
+                maskslits=self.maskslits,show_peaks=self.show)
+
+        # If there are objects, do 2nd round of global_skysub, local_skysub_extract, flexure, geo_motion
+        if self.nobj > 0:
+            if not self.std_redux:
+                # Global sky subtraction second pass. Uses skymask from object finding
+                global_sky = self.sciI.global_skysub(self.caliBrate.tilts_dict['tilts'],
+                                                     skymask=self.skymask, maskslits=self.maskslits, show=self.show)
+            skymodel, objmodel, ivarmodel, outmask, sobjs = \
+                self.sciI.local_skysub_extract(self.sobjs_obj, self.caliBrate.mswave, model_noise=(not self.ir_redux),
+                                               std = self.std_redux,maskslits=self.maskslits, show_profile=self.show,
+                                               show=self.show)
+
+            # Purge out the negative objects if this was a near-IR reduction
+            if self.ir_redux:
+                sobjs.purge_neg()
+
+            # Flexure correction if this is not a standard star
+            if not self.std_redux:
+                self.flexure_correct(sobjs, self.maskslits)
+            vel_corr = self.helio_correct(sobjs, self.maskslits, frames[0], self.obstime)
+
+        else:
+            # Print status message
+            msgs_string = 'No objects to extract for target {:s}'.format(self.fitstbl['target'][frames[0]]) + msgs.newline()
+            msgs_string += 'On frames:' + msgs.newline()
+            for iframe in frames:
+                msgs_string += '{0:s}'.format(self.fitstbl['filename'][iframe]) + msgs.newline()
+            msgs.warn(msgs_string)
+            # set to first pass global sky
+            skymodel = initial_sky
+            objmodel = np.zeros_like(sciimg)
+            # Set to sciivar. Could create a model but what is the point?
+            ivarmodel = np.copy(sciivar)
+            # Set to inmask in case on objects were found
+            outmask = self.sciI.mask
+            # empty specobjs object from object finding
+            sobjs = self.sobjs_obj
+            vel_corr = None
+
+        return sciimg, sciivar, skymodel, objmodel, ivarmodel, outmask, sobjs, vel_corr
+
+
+    def find_objects(self, image, std=False, ir_redux=False, std_trace=None, snr_trim=False, maskslits=None,
```

Comment:
As I noted above.  Why not put this as separate methods
in children of the `PypeIt` class

Reply:
Moved find_objects method to children of PypeIt class.